design-proposals/dynamic-primary-pod-nic-name.md

```diff
@@ -0,0 +1,481 @@
+In case the field will be empty, virt-handler and virt-launcher will fall back to the existing naming scheme calculation based on the VMI.spec.
+
+
+#### API Example
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: myvmi
+spec:
+  ...
+status:
+  ...
+  primaryPodInterfaceName: custom-iface
+  interfaces:
+    - infoSource: 'domain, guest-agent, multus-status'
+      interfaceName: eth0
+      ipAddress: 10.131.0.29
+      ipAddresses:
+        - 10.131.0.29
+      mac: '52:54:00:c0:35:78'
+      name: passtnet
+      queueCount: 1
+  ...
+```
+
+#### Pros
+
+- Manual configuration is not required
+- Provides backward compatibility
+
+#### Cons
+
+- Adds API
+- Exposes an implementation detail to VM owners which is probably not useful for them
+- There is no correlation with the relevant interface (which might not even exist in case the VM is not connected to the pod network).
+- This field is not relevant in case the VM is using only secondary networks
+
+### Alternative 3 - Use annotations instead of K8s API
+
+A new annotation will be added to the VirtualMachineInstance object.
+It could represent either alternative 1 or 2 in annotation form.
+
+#### Pros
+
+- Provides backward compatibility
+
+#### Cons
+
+- Since information is going to be passed between KubeVirt's components it is not ideal to use semiformal API. 
+- Annotations usually flow from VirtualMachineInstance to Pod and not vice versa.
+
+### Alternative 4 - Use the existing Downward API mechanism
+
+Currently, the VMI controller sets the `kubevirt.io/network-info` annotation on the virt-launcher pod in two cases:
+
+1. The VMI has an SR-IOV NIC.
+2. The VMI uses a network binding plugin that is configured to use downward API.
+
+The annotation's value is derived fom Multus' network-status annotation and is injected into the virt-launcher pod
+using K8s' [Downward API](https://kubernetes.io/docs/concepts/workloads/pods/downward-api/) mechanism.
+
+A new field will be added to the `NetworkInfo` struct:
+
+```go
+type NetworkInfo struct {
+Interfaces []Interface `json:"interfaces,omitempty"`
+PrimaryPodInterfaceName string `json:primaryPodInterfaceName,omitempty`
+}
+```
+
+A new mechanism should be added between virt-handler and virt-launcher to share this information, because:
+
+1. The primary pod interface name is also required by virt-handler
+2. The virt-handler does not have access to pod objects
+
+virt-handler will poll the virt-launcher's file-system for the `/etc/podinfo` file.
+virt-handler will not be able to start the network setup until it successfully read and parsed the content of the file above.
+
+#### Cons
+
+- All virt-launcher pods will have to use the downward API mechanism instead of just the subset using it today
+- Complicates the network setup and overall VM startup sequence
+
+## Specifying the primary pod interface name
+
```

Comment:
You better give here a sentence about this option main idea: Decide what the name is from the Kubevirt side.